From 5a5e47ceedaeb69b1bf580f553d8f5eb23e60072 Mon Sep 17 00:00:00 2001
From: Patrick Yavitz <pyavitz@xxxxx.com>
Date: Sat, 19 Aug 2023 17:00:22 -0400
Subject: [PATCH] arch: arm64: dts: amlogic: meson-g12b-sm1-odroid

Add meson64-reboot driver to all boards
Add uart_A uart_AO_B uart_B uart_C where appropriate

Signed-off-by: Patrick Yavitz <pyavitz@xxxxx.com>
---
 .../dts/amlogic/meson-g12b-odroid-n2-plus.dts |  8 +++-
 .../boot/dts/amlogic/meson-g12b-odroid-n2.dts |  7 ++++
 .../dts/amlogic/meson-g12b-odroid-n2l.dts     |  9 ++++-
 .../boot/dts/amlogic/meson-g12b-odroid.dtsi   | 37 +++++++++++++++++-
 .../boot/dts/amlogic/meson-sm1-odroid-c4.dts  | 16 ++++++++
 .../boot/dts/amlogic/meson-sm1-odroid-hc4.dts | 23 +++++++++++
 .../boot/dts/amlogic/meson-sm1-odroid.dtsi    | 38 +++++++++++++++++++
 7 files changed, 135 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2-plus.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2-plus.dts
index ce1198ad34e4..72cd79119def 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2-plus.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2-plus.dts
@@ -15,6 +15,13 @@ / {
 	model = "Hardkernel ODROID-N2Plus";
 };
 
+&gpu_opp_table {
+	opp-999999984 {
+		opp-hz = /bits/ 64 <999999984>;
+		opp-microvolt = <800000>;
+	};
+};
+
 &vddcpu_a {
 	regulator-min-microvolt = <680000>;
 	regulator-max-microvolt = <1040000>;
@@ -28,4 +35,3 @@ &vddcpu_b {
 
 	pwms = <&pwm_AO_cd 1 1500 0>;
 };
-
diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts
index a198a91259ec..b77b2c425d9e 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts
@@ -13,3 +13,10 @@ / {
 	compatible = "hardkernel,odroid-n2", "amlogic,s922x", "amlogic,g12b";
 	model = "Hardkernel ODROID-N2";
 };
+
+&gpu_opp_table {
+	opp-999999984 {
+		opp-hz = /bits/ 64 <999999984>;
+		opp-microvolt = <800000>;
+	};
+};
diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2l.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2l.dts
index 70919f40d597..037aa4fe0de0 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2l.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2l.dts
@@ -13,6 +13,10 @@ / {
 	compatible = "hardkernel,odroid-n2l", "amlogic,s922x", "amlogic,g12b";
 	model = "Hardkernel ODROID-N2L";
 
+	aliases {
+		/delete-property/ rtc0;
+	};
+
 	sound {
 		compatible = "amlogic,axg-sound-card";
 		model = "ODROID-N2L";
@@ -46,7 +50,6 @@ sound {
 		assigned-clock-rates = <294912000>,
 				       <270950400>,
 				       <393216000>;
-		status = "okay";
 
 		dai-link-0 {
 			sound-dai = <&frddr_a>;
@@ -102,6 +105,10 @@ &eth_phy {
 	status = "disabled";
 };
 
+&i2c3 {
+	/delete-node/ rtc@51;
+};
+
 &vddcpu_a {
 	regulator-min-microvolt = <680000>;
 	regulator-max-microvolt = <1040000>;
diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid.dtsi b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid.dtsi
index 9e12a34b2840..b549852d3d40 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid.dtsi
@@ -12,6 +12,9 @@
 / {
 	aliases {
 		serial0 = &uart_AO;
+		serial1 = &uart_A;
+		serial2 = &uart_B;
+		serial3 = &uart_C;
 		ethernet0 = &ethmac;
 		rtc1 = &vrtc;
 	};
@@ -20,6 +23,16 @@ chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
+	reboot: meson64-reboot {
+		compatible = "meson64,reboot";
+		sys_reset = <0x84000009>;
+		sys_poweroff = <0x84000008>;
+
+		sd-vmmc = <&gpio_ao GPIOAO_8 GPIO_ACTIVE_HIGH>;
+		sd-vqen = <&gpio_ao GPIOAO_3 GPIO_ACTIVE_HIGH>;
+		sd-vqsw = <&gpio_ao GPIOAO_9 GPIO_ACTIVE_HIGH>;
+	};
+
 	memory@0 {
 		device_type = "memory";
 		reg = <0x0 0x0 0x0 0x40000000>;
@@ -354,7 +367,11 @@ &sd_emmc_b {
 
 	bus-width = <4>;
 	cap-sd-highspeed;
-	max-frequency = <50000000>;
+	max-frequency = <200000000>;
+	sd-uhs-sdr12;
+	sd-uhs-sdr25;
+	sd-uhs-sdr50;
+	sd-uhs-sdr104;
 	disable-wp;
 
 	cd-gpios = <&gpio GPIOC_6 GPIO_ACTIVE_LOW>;
@@ -430,12 +447,30 @@ &toddr_c {
 	status = "okay";
 };
 
+&uart_A {
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart_a_pins>;
+};
+
 &uart_AO {
 	status = "okay";
 	pinctrl-0 = <&uart_ao_a_pins>;
 	pinctrl-names = "default";
 };
 
+&uart_B {
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart_b_pins>;
+};
+
+&uart_C {
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart_c_pins>;
+};
+
 &usb {
 	status = "okay";
 };
diff --git a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-c4.dts b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-c4.dts
index d04768a66bfe..b51498ce93b3 100644
--- a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-c4.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-c4.dts
@@ -56,3 +56,19 @@ hub_3_0: hub@2 {
 &ir {
 	linux,rc-map-name = "rc-odroid";
 };
+
+&external_phy {
+	/delete-property/	interrupts;
+	/delete-property/	interrupts-parent;
+};
+
+&gpu_opp_table {
+	opp-999999984 {
+		opp-hz = /bits/ 64 <999999984>;
+		opp-microvolt = <800000>;
+	};
+};
+
+&vddcpu {
+	regulator-max-microvolt = <1030000>;
+};
diff --git a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts
index 74088e7280fe..9c5701a1fc49 100644
--- a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid-hc4.dts
@@ -142,3 +142,26 @@ &usb {
 	phys = <&usb2_phy0>, <&usb2_phy1>;
 	phy-names = "usb2-phy0", "usb2-phy1";
 };
+
+&external_phy {
+	/delete-property/	interrupts;
+	/delete-property/	interrupts-parent;
+};
+
+&fan0 {
+	fan-supply = <&vcc_5v>;
+	interrupt-parent = <&gpio_intc>;
+	interrupts = <84 IRQ_TYPE_EDGE_FALLING>;
+	pulses-per-revolutions = <2>;
+};
+
+&gpu_opp_table {
+	opp-999999984 {
+		opp-hz = /bits/ 64 <999999984>;
+		opp-microvolt = <800000>;
+	};
+};
+
+&vddcpu {
+	regulator-max-microvolt = <1030000>;
+};
diff --git a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid.dtsi b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid.dtsi
index 2fce44939f45..aa6bdb84817a 100644
--- a/arch/arm64/boot/dts/amlogic/meson-sm1-odroid.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-sm1-odroid.dtsi
@@ -11,6 +11,10 @@
 / {
 	aliases {
 		serial0 = &uart_AO;
+		serial1 = &uart_A;
+		serial2 = &uart_B;
+		serial3 = &uart_C;
+		serial4 = &uart_AO_B;
 		ethernet0 = &ethmac;
 	};
 
@@ -18,6 +22,16 @@ chosen {
 		stdout-path = "serial0:115200n8";
 	};
 
+	reboot: meson64-reboot {
+		compatible = "meson64,reboot";
+		sys_reset = <0x84000009>;
+		sys_poweroff = <0x84000008>;
+
+		sd-vqen = <&gpio_ao GPIOE_2 GPIO_ACTIVE_HIGH>;
+		sd-vqsw = <&gpio_ao GPIOAO_6 GPIO_ACTIVE_HIGH>;
+		sd-vmmc = <&gpio_ao GPIOAO_3 GPIO_ACTIVE_HIGH>;
+	};
+
 	memory@0 {
 		device_type = "memory";
 		reg = <0x0 0x0 0x0 0x40000000>;
@@ -450,12 +464,36 @@ &tohdmitx {
 	status = "okay";
 };
 
+&uart_A {
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart_a_pins>;
+};
+
 &uart_AO {
 	status = "okay";
 	pinctrl-0 = <&uart_ao_a_pins>;
 	pinctrl-names = "default";
 };
 
+&uart_AO_B {
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart_ao_b_8_9_pins>;
+};
+
+&uart_B {
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart_b_pins>;
+};
+
+&uart_C {
+	status = "disabled";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart_c_pins>;
+};
+
 &usb {
 	status = "okay";
 	vbus-supply = <&usb_pwr_en>;
-- 
2.39.2

